action/library/src/main/java/io/knotx/fragments/action/library/cache/CacheAction.java

```diff
@@ -0,0 +1,135 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.fragments.action.library.cache;
+
+import static io.knotx.commons.validation.ValidationHelper.checkArgument;
+import static io.knotx.fragments.action.library.helper.FragmentPlaceholders.buildSourceDefinitions;
+import static io.knotx.fragments.api.FragmentResult.fail;
+
+import io.knotx.commons.cache.Cache;
+import io.knotx.fragments.action.api.Action;
+import io.knotx.fragments.action.api.SingleAction;
+import io.knotx.fragments.action.api.log.ActionLogLevel;
+import io.knotx.fragments.action.library.cache.operations.CacheActionLogger;
+import io.knotx.fragments.action.library.cache.operations.CacheLookup;
+import io.knotx.fragments.action.library.cache.operations.CacheStore;
+import io.knotx.fragments.action.library.exception.ActionConfigurationException;
+import io.knotx.fragments.api.FragmentContext;
+import io.knotx.fragments.api.FragmentResult;
+import io.knotx.reactivex.fragments.api.FragmentOperation;
+import io.knotx.server.common.placeholders.PlaceholdersResolver;
+import io.reactivex.Maybe;
+import io.reactivex.Single;
+import org.apache.commons.lang3.StringUtils;
+
+public class CacheAction implements SingleAction {
+
+  private final boolean failWhenLookupFails;
+  private final boolean failWhenStoreFails;
+
+  private final ActionLogLevel logLevel;
+  private final String keySchema;
+  private final String alias;
+
+  private final Action doAction;
+
+  private final CacheLookup lookup;
+  private final CacheStore store;
+
+  public static CacheAction create(Cache cache, CacheActionOptions options, String alias,
+      Action doAction) {
+    checkArgument(StringUtils.isBlank(options.getPayloadKey()),
+        () -> new ActionConfigurationException(alias,
+            "CacheAction requires payloadKey value in configuration."));
+    checkArgument(StringUtils.isBlank(options.getCacheKey()),
+        () -> new ActionConfigurationException(alias,
+            "CacheAction requires cacheKey value in configuration."));
+    checkArgument(doAction == null, () -> new ActionConfigurationException(alias,
+        "CacheAction requires doAction configured but none provided"));
+    return new CacheAction(
+        alias,
+        options.getCacheKey(),
+        doAction,
+        ActionLogLevel.fromConfig(options.getLogLevel(), ActionLogLevel.ERROR),
+        new CacheLookup(cache, options.getPayloadKey()),
+        new CacheStore(cache, options.getPayloadKey()),
+        options.isFailWhenLookupFails(),
+        options.isFailWhenStoreFails()
+    );
+  }
+
+  public CacheAction(String alias, String keySchema, Action doAction, ActionLogLevel logLevel,
```

Comment:
The idea is that by default `CacheActionFactory` uses factory method, but unit tests use constructor directly. Some sugar as there is no DI so far :)